test/jdk/javax/swing/JProgressBar/TestProgressBarBorder.java

```diff
@@ -0,0 +1,104 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.File;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import javax.imageio.ImageIO;
+import javax.swing.JComponent;
+import javax.swing.JProgressBar;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+import static java.awt.image.BufferedImage.TYPE_INT_RGB;
+
+/*
+ * @test
+ * @bug 8224261
+ * @key headful
+ * @library ../regtesthelpers
+ * @build Util
+ * @summary Verifies JProgressBar border is not painted when border
+ *          painting is set to false
+ * @run main TestProgressBarBorder
+ */
+
+public class TestProgressBarBorder {
+    private static JProgressBar progressBar;
+    private static volatile BufferedImage withBorder = null;
+    private static volatile BufferedImage withoutBorder = null;
+
+    public static void main(String[] args) throws Exception {
+        boolean equal;
+        for (UIManager.LookAndFeelInfo laf :
+                UIManager.getInstalledLookAndFeels()) {
+            if (!laf.getName().contains("Nimbus") && !laf.getName().contains("GTK")) {
+                continue;
+            }
+            System.out.println("Testing LAF: " + laf.getName());
+            SwingUtilities.invokeAndWait(() -> {
+                setLookAndFeel(laf);
+                initProgressBar();
+                progressBar.setBorderPainted(true);
+                withBorder = paintToImage(progressBar);
+                progressBar.setBorderPainted(false);
+                withoutBorder = paintToImage(progressBar);
+            });
+            equal = Util.compareBufferedImages(withBorder, withoutBorder);
+            if (equal) {
+                ImageIO.write(withBorder, "png", new File("withBorder.png"));
+                ImageIO.write(withoutBorder, "png", new File("withoutBorder.png"));
+                throw new RuntimeException("JProgressBar border is painted when border\n" +
+                        " painting is set to false");
```

Comment:
```suggestion
                throw new RuntimeException("JProgressBar border is painted when border " +
                        "painting is set to false");
```
You should probably remove the line break from the error message.